connectors/services/sync.py

```diff
@@ -111,22 +118,26 @@ async def _run(self):
         es = ElasticServer(self.es_config)
         try:
             while self.running:
+                syncs = ConcurrentTasks(max_concurrency=self.concurrent_syncs)
                 try:
                     logger.debug(f"Polling every {self.idling} seconds")
                     query = self.connectors.build_docs_query(
                         native_service_types, connectors_ids
                     )
-
                     async for connector in self.connectors.get_all_docs(query=query):
-                        await self._one_sync(connector, es, sync_now)
+                        await syncs.put(
```

Comment:
should we call `put` when a real sync happens?

Reply:
not sure what you mean.  

- If there's room, `put` will push the task into the pool, and also start it.  
- if there's no run, this blocks until there's room

Reply:
Now you `put` `_one_sync` into the `ConcurrentTasks`. Inside `_one_sync` there are still some checks, e.g. whether the connector is configured to sync, and whether the connector is scheduled/forced to sync. If all checks pass, it will sync eventually by calling `connector.sync()`(will be refactored to a separate class `SyncJobRunner`), and this should be the task to `put` into `ConcurrentTasks`